src/token.rs

```diff
@@ -153,6 +172,36 @@ fn make_token<P: Serialize + Deserialize + 'static>(subject: &str,
                              }))
 }
 
+/// Verify that the service requested for is allowed in the configuration
+fn verify_service(config: &Configuration, service: &str) -> Result<(), Error> {
+    if !config.audience.contains(&FromStr::from_str(service)?) {
```

Comment:
Is this correct? If my audience is `http://www.example.com` and my service is `http://www.e` it returns `Ok(())`?